internal/logic_handlers/organizationlogic/service_test.go

```diff
@@ -0,0 +1,680 @@
+
+			updateOrganizationStruct := tt.mutations(t, _dependencies.Db)
+
+			organization, err := svc.UpdateOrganization(tt.ctx, updateOrganizationStruct)
+			tt.assertions(t, updateOrganizationStruct, organization, err)
+		})
+	}
+}
+
+type testCaseCreateOrganization struct {
+	name            string
+	ctx             context.Context
+	getDependencies func(t *testing.T) (*dependencies, func(ignoreErrors ...bool))
+	mutations       func(t *testing.T, db *sqlx.DB) (org *model.CreateOrganization)
+	assertions      func(t *testing.T, organization *model.Organization, err error)
+}
+
+func getTestCasesCreateOrganization() []testCaseCreateOrganization {
+	return []testCaseCreateOrganization{
+		{
+			name:            "success",
+			getDependencies: getConcreteDependencies,
+			ctx:             context.TODO(),
+			assertions: func(t *testing.T, organization *model.Organization, err error) {
+				require.NoError(t, err, "unexpected error")
+				require.NotNil(t, organization, "unexpected nil organization")
+
+				require.Equal(t, 1, 1, "unexpected nil organization")
+				require.NotEmpty(t, "Demby", "unexpected empty organization name")
+				require.Equal(t, strconv.Itoa(1), organization.CreatedBy, "unexpected empty Created_by")
+			},
+			mutations: func(t *testing.T, db *sqlx.DB) (org *model.CreateOrganization) {
+				user, err := mysqlmodel.Users().One(context.TODO(), db)
+				require.NoError(t, err, "error fetching one user record")
+
+				createOrg := model.CreateOrganization{
+					Name:   "Demby",
+					UserId: user.ID,
+				}
+
+				return &createOrg
+			},
+		},
+		{
+			name:            "fail-internal-error",
+			getDependencies: getConcreteDependencies,
+			ctx:             context.TODO(),
+			assertions: func(t *testing.T, organization *model.Organization, err error) {
+				assert.Error(t, err, "unexpected error")
+				assert.Nil(t, organization, "unexpected nil organization")
+			},
+			mutations: func(t *testing.T, db *sqlx.DB) (org *model.CreateOrganization) {
+				testhelper.DropTable(t, db, mysqlmodel.TableNames.Organization)
+				return nil
+			},
+		},
+		{
+			name:            "fail-invalid-args",
+			getDependencies: getConcreteDependencies,
+			ctx:             context.TODO(),
+			assertions: func(t *testing.T, organization *model.Organization, err error) {
+				assert.Error(t, err, "unexpected error")
+				assert.Nil(t, organization, "unexpected nil organization")
+			},
+			mutations: func(t *testing.T, db *sqlx.DB) (org *model.CreateOrganization) {
+				createOrg := &model.CreateOrganization{
+					Name:   "Demby",
+					UserId: rand.Int(),
+				}
+
+				return createOrg
+			},
+		},
+		{
+			name:            "fail-unique",
+			getDependencies: getConcreteDependencies,
+			ctx:             context.TODO(),
+			assertions: func(t *testing.T, organization *model.Organization, err error) {
+				assert.Error(t, err, "unexpected error")
+				assert.Nil(t, organization, "unexpected nil organization")
```

Comment:
assert.Contains(t, err.Error(), "entry already exists") // try to apply something like this

// Lets see if you can figure this out

Reply:
Do we need to apply assert.Contains on all the test cases?

Reply:
assert.Contains is now used on the test cases.